addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/connection/SerialConnector.java

```diff
@@ -0,0 +1,371 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nibeheatpump.internal.connection;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InterruptedIOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TooManyListenersException;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.apache.commons.io.IOUtils;
+import org.openhab.binding.nibeheatpump.internal.NibeHeatPumpException;
+import org.openhab.binding.nibeheatpump.internal.config.NibeHeatPumpConfiguration;
+import org.openhab.binding.nibeheatpump.internal.message.MessageFactory;
+import org.openhab.binding.nibeheatpump.internal.message.ModbusReadRequestMessage;
+import org.openhab.binding.nibeheatpump.internal.message.ModbusWriteRequestMessage;
+import org.openhab.binding.nibeheatpump.internal.message.NibeHeatPumpMessage;
+import org.openhab.binding.nibeheatpump.internal.protocol.NibeHeatPumpProtocol;
+import org.openhab.binding.nibeheatpump.internal.protocol.NibeHeatPumpProtocolContext;
+import org.openhab.binding.nibeheatpump.internal.protocol.NibeHeatPumpProtocolDefaultContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPort;
+import gnu.io.CommPortIdentifier;
+import gnu.io.NoSuchPortException;
+import gnu.io.PortInUseException;
+import gnu.io.SerialPort;
+import gnu.io.SerialPortEvent;
+import gnu.io.SerialPortEventListener;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * Connector for serial port communication.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class SerialConnector extends NibeHeatPumpBaseConnector {
+
+    private static final Logger logger = LoggerFactory.getLogger(SerialConnector.class);
+
+    private InputStream in = null;
+    private OutputStream out = null;
+    private SerialPort serialPort = null;
+    private Thread readerThread = null;
+    private NibeHeatPumpConfiguration conf = null;
+
+    private List<byte[]> readQueue = new ArrayList<byte[]>();
+    private List<byte[]> writeQueue = new ArrayList<byte[]>();
+
+    public SerialConnector() {
+
+        logger.debug("Nibe heatpump Serial Port message listener created");
+    }
+
+    @Override
+    public void connect(NibeHeatPumpConfiguration configuration) throws NibeHeatPumpException {
+        if (isConnected()) {
+            return;
+        }
+
+        conf = configuration;
+        try {
+            CommPortIdentifier portIdentifier = CommPortIdentifier.getPortIdentifier(conf.serialPort);
+
+            CommPort commPort = portIdentifier.open(this.getClass().getName(), 2000);
+
+            serialPort = (SerialPort) commPort;
+            serialPort.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+
+            serialPort.enableReceiveThreshold(1);
+            serialPort.disableReceiveTimeout();
```

Comment:
That will enable blocking read. Serial port is closed during binding dispose, which terminate also blocking read method